src/webui/api/apicontroller.cpp

```diff
@@ -79,20 +79,29 @@ void APIController::requireParams(const QVector<QString> &requiredParams) const
 
 void APIController::setResult(const QString &result)
 {
-    m_result = result;
+    m_result.data = result;
 }
 
 void APIController::setResult(const QJsonArray &result)
 {
-    m_result = QJsonDocument(result);
+    m_result.data = QJsonDocument(result);
 }
 
 void APIController::setResult(const QJsonObject &result)
 {
-    m_result = QJsonDocument(result);
+    m_result.data = QJsonDocument(result);
 }
 
-void APIController::setResult(const QByteArray &result)
+void APIController::setResult(const QByteArray &result, const QString &mimeType, const QString &filename)
 {
-    m_result = result;
+    m_result.data = result;
+    m_result.mimeType = mimeType;
+    m_result.filename = filename;
+}
+
+void APIResult::clear()
```

Comment:
Despite I approved, I would hope to see it moved above the `APIController` constructor so it has the same layout like in the header file.

Reply:
> I would hope to see it moved above the `APIController` constructor so it has the same layout like in the header file.

Changed.